atlp-rwanda/e-comm-team-techsmith-bn
# #184752086 Issuing JWT after successful user registration

## What does this PR do?

This PR serves the functionality for user signup, and receiving a [jwt](https://jwt.io/) after successful registration.

## Description of Task to be completed?

#### In the process of user registration, this PR works as follows:

- If a user submits all required inputs, they should be registered successfully, and receive a [jwt](https://jwt.io/) for authorization.
- The jwt is issued as cookie, named `Authorization`.

#### If a user makes a bad request, they receive any of the following feedbacks:

- If a user submits a password shorter than 8 characters, they receive an error message: *Password must be at least 8 characters long*.

- If a user submits an invalid password or email, they receive an error message: *Invalid email* or *Password must contain at least one letter and one number*.

## How should this be manually tested?

1. Clone the repo using `git clone` [repo](https://github.com/atlp-rwanda/e-comm-team-techsmith-bn)

2. Run `npm install` from the directory root

3. Create a `.env` to store sensitive credentials used in this application. Make sure to check `.env.example` to find all environment variables the application requires to be operational.

4. Refer to the user registration endpoint by navigating to `http://localhost:{port}/api/docs` using the browser.

5. Use postman to test the end the user registration endpoint: `http://localhost:{port}/api/user/signup`.

- `port` is a port on which the application is running, defined in `.env`.

## Any background context you want to provide?

Given that a user has to be issued a jwt immediately after registration, the token is being signed using a user __email__ instead of using their unique id.

## What are the relevant pivotal tracker/Trello stories?

- [#184752086](https://www.pivotaltracker.com/story/show/184752086)
